cmake/VASTPackage.cmake

```
@@ -47,4 +45,45 @@ set(CPACK_DEBIAN_PACKAGE_CONTROL_EXTRA
     "${CMAKE_CURRENT_SOURCE_DIR}/scripts/debian/prerm")
 set(CPACK_DEBIAN_PACKAGE_DEBUG ON)
 
+# For the static binary builds it doesn't make much sense to install development
+# utilities, so we never do so.
+if (VAST_ENABLE_STATIC_EXECUTABLE)
+  get_cmake_property(CPACK_COMPONENTS_ALL COMPONENTS)
+  list(REMOVE_ITEM CPACK_COMPONENTS_ALL "Development" "Unspecified")
+endif ()
+
+# Create only a single package with all enabled install components rather than
+# creating one package per install component that potentially duplicates the
+# files. We should reconsider this in the future when we want to offer separate
+# runtime and development Debian packages.
+set(CPACK_COMPONENTS_GROUPING ALL_COMPONENTS_IN_ONE)
+
+# Enable the component-based packages for the Debian and the Archive CPack
+# generators. Sadly, this has to be done 1-by-1, and there does not appear to be
+# an exhaustive list of generators that support this, so we only opt-in for the
+# two common cases.
+set(CPACK_DEBIAN_COMPONENT_INSTALL ON)
```

CMake is really creative in finding ways to trip you up
```suggestion
set(CPACK_DEB_COMPONENT_INSTALL ON)
```